src/patches.js

```diff
@@ -6,50 +6,92 @@ export function generatePatches(state, basePath, patches, inversePatches) {
         : generateObjectPatches(state, basePath, patches, inversePatches)
 }
 
-function generateArrayPatches(state, basePath, patches, inversePatches) {
-    const {base, copy, assigned} = state
-    const minLength = Math.min(base.length, copy.length)
+function _generateArrayPatches(
+    base,
+    copy,
+    assigned,
+    basePath,
+    patches,
+    inversePatches
+) {
+    if (copy.length < base.length) {
+        return _generateArrayPatches(
```

Comment:
Not a fan of the underscore. Since you're just swapping arguments, maybe do this instead:
```js
;[copy, base] = [base, copy]
;[patches, inversePatches] = [inversePatches, patches]
```